demos/wstrust/shared/src/main/java/org/jboss/test/ws/jaxws/samples/wsse/policy/trust/shared/UsernameTokenCallbackHandler.java

```diff
@@ -0,0 +1,173 @@
+/**
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jboss.test.ws.jaxws.samples.wsse.policy.trust.shared;
+
+import java.io.IOException;
+import java.util.Map;
+
+import javax.security.auth.callback.Callback;
+import javax.security.auth.callback.CallbackHandler;
+import javax.security.auth.callback.UnsupportedCallbackException;
+
+import org.apache.cxf.helpers.DOMUtils;
+import org.apache.cxf.message.Message;
+import org.apache.cxf.ws.security.SecurityConstants;
+import org.apache.cxf.ws.security.trust.delegation.DelegationCallback;
+import org.apache.wss4j.dom.WSConstants;
+import org.apache.wss4j.dom.message.token.UsernameToken;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.ls.DOMImplementationLS;
+import org.w3c.dom.ls.LSSerializer;
+
+/**
+ * A utility to provide the 3 different input parameter types for jaxws property
+ * "ws-security.sts.token.act-as" and "ws-security.sts.token.on-behalf-of".
+ * This implementation obtains a username and password via the jaxws property
+ * "ws-security.username" and "ws-security.password" respectively, as defined
+ * in SecurityConstants.  It creates a wss UsernameToken to be used as the
+ * delegation token.
+ *
+ */
+
+public class UsernameTokenCallbackHandler implements CallbackHandler {
+
+	public void handle(Callback[] callbacks)
+			throws IOException, UnsupportedCallbackException {
+		for (int i = 0; i < callbacks.length; i++) {
+			if (callbacks[i] instanceof DelegationCallback) {
+				DelegationCallback callback = (DelegationCallback) callbacks[i];
+				Message message = callback.getCurrentMessage();
+
+				String username = (String) message.getContextualProperty(SecurityConstants.USERNAME);
+				String password = (String) message.getContextualProperty(SecurityConstants.PASSWORD);
+				if (username != null) {
```

Comment:
What is it is not null, but empty? Should this be checked actually? Same for the similar  cases below.

Reply:
A check is not needed.  The callback handler will flag issues.